vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/scatter_plotly.tsx

```diff
@@ -0,0 +1,232 @@
+
+  const updateValue = (newValue: any, key: string, prevValues = value) => {
+    prevValues[key] = newValue;
+    onChange(some(prevValues));
+  };
+
+  // Component Setups
+  const string_input = (
+    key: string = "filler", value: string | number | boolean = "filler",
+    label: string = 'hello') => {
+      return (
+        <TextInput
+          key={key}
+          header={label}
+          value={value}
+          onChange={(newValue: string) => updateValue(newValue, key)}
+        />
+      )
+    };
+
+  const nestable_dropdown_input = (
+    key: string = "filler", value: Maybe<string>,
+    label: string, options: DataEnvelope<null>) => {
+      
+      return(
+        <div key={key}>
+          {label}
+          <NestedSelectAutocompleteInput
+            data={{options}}
+            value={value}
+            onChange={(val) => updateValue(val, key)}
+          />
+        </div>
+      )
+    }
+  
+  const dropdown_input = (
+    key: string = "filler", value: Maybe<string>,
+    label: string, defaultValue: string | null = null, options: string[]) => {
+      
+      return(
+        <div key={key}>
+          {label}
+          <DropdownAutocomplete
+            list={options}
+            defaultValue={defaultValue}
+            value={value}
+            onChange={(val: string) => updateValue(val, key)}
+          />
+        </div>
+      )
+    }
+      
+    
+
+  const checkbox_input = (
+    key: string = "filler", value: boolean = false,
+    label: string) => {
+
+      return(
+          <div key={key}>
+            {label}
+            <input id={key} type='checkbox'
+                  checked={value}
+                  onChange={() => updateValue(!value, key)} />
+          </div>
+      )
+    }
+
+  const slider_input = (
+    key: string = "filler", value: number,
+    label: string, min: number = 0.1, max: number = 20) => {
+
+      return(
+          <div key={key}>
+            {label}
+            <Slider
+              value={value}
+              onChange={(event, newValue) => updateValue(newValue as number, key)}
+              min={min}
+              max={max}
+              valueLabelDisplay="auto"
+            />
+          </div>
+      )
+    }
+
+  const extra_inputs: DataEnvelope<any[]> = useMemo(() => {
+    return {
+      // label, then for any extras
+      'plot_title': ['Plot Title'],
+      'legend_title': ['Legend Title'],
+      'xlab': ['X-Axis Title'],
+      'ylab': ['Y-Axis Title'],
+      'x_by': ['X-Axis Data', options],
+      'y_by': ['Y-Axis Data', options],
+      'color_by': ['Color Points By', options],
+      'color_order': ['Point render order', null, ['increasing', 'decreasing', 'unordered']],
+      'order_when_continuous_color': ['Follow selected render ordering when color is continuous?'],
+      'size': ['Point Size', 0.1, 50]
+    }
+  }, [options]);
+
+  // Component set constructor
+  const component_use = (key: string, value: any, extra_inputs: any) => {
```

Comment:
Very interesting.  The structure of this component is as a nested set of other configurable inputs.  This, works for now, but I imagine is fragile and prone to breakage / difficulty to maintain.  Later, when we have the behavior in a more stable place, let's imagine how we can deconstruct some of the roles of this component into composable separate pieces.

Reply:
Yea, I hate it myself, and that's definitely the plan lol.